skymarshal/skyserver/skyserver_test.go

```diff
@@ -370,18 +370,54 @@ var _ = Describe("Sky Server API", func() {
 
 						fakeTokenVerifier.VerifyReturns(fakeVerifiedClaims, nil)
 						fakeTokenIssuer.IssueReturns(fakeOAuthToken, nil)
+					})
 
-						state, _ := json.Marshal(map[string]string{
-							"redirect_uri": "http://example.com",
+					Context("when redirect URI is http://example.com", func() {
+						BeforeEach(func() {
+							state, _ := json.Marshal(map[string]string{
+								"redirect_uri": "http://example.com",
+							})
+
+							stateToken := base64.StdEncoding.EncodeToString(state)
+							stateCookie = &http.Cookie{Name: "skymarshal_state", Value: stateToken}
+							reqPath = "/sky/callback?code=some-code&state=" + stateToken
 						})
 
-						stateToken := base64.StdEncoding.EncodeToString(state)
-						stateCookie = &http.Cookie{Name: "skymarshal_state", Value: stateToken}
-						reqPath = "/sky/callback?code=some-code&state=" + stateToken
+						It("errors", func() {
+							Expect(response.StatusCode).To(Equal(http.StatusBadRequest))
+						})
 					})
 
-					It("errors", func() {
-						Expect(response.StatusCode).To(Equal(http.StatusBadRequest))
+					Context("when redirect URI is https:example.com", func() {
+						BeforeEach(func() {
+							state, _ := json.Marshal(map[string]string{
+								"redirect_uri": "https:google.com",
```

Comment:
Does it matter if these references are "https:google.com" or "https://google.com"? 🤔

Reply:
yes it matters. "https://google.com" is a valid URL that we know it wont cause problem. But "https:google.com" is the one we want to test in this scenario.

Reply:
My mistake - I read the code incorrectly. Thanks for clearing that up for me!